feat: add appium docker image build API 34

Bilb opened the pull request.

Bilb committed:
feat: add appium docker image build API 34

jagerman commented on ci/rebuild-docker-images.py:
```diff
@@ -485,6 +488,98 @@ def playwright_build(distro, arch):
+# Install required Android CMD-line tools
+#============================================
+RUN wget https://dl.google.com/android/repository/$ANDROID_CMD -P /tmp && \
+              unzip -d $ANDROID_SDK_ROOT /tmp/$ANDROID_CMD && \
```
You should add a `&& rm -f /tmp/$ANDROID_CMD` after unzipping (so that the temporary zip file doesn't have to get put into the image.)

jagerman commented on ci/rebuild-docker-images.py:
```diff
@@ -485,6 +488,98 @@ def playwright_build(distro, arch):
+
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get update
```
```suggestion
RUN apt-get -o=Dpkg::Use-Pty=0 -q update
```
This makes sure packages won't be able to stop and ask for input.

jagerman commented on ci/rebuild-docker-images.py:
```diff
@@ -485,6 +488,98 @@ def playwright_build(distro, arch):
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get update
+RUN apt install -y ca-certificates curl git cpu-checker supervisor vim bash wget unzip xvfb x11vnc fluxbox xterm novnc net-tools htop libpulse-dev libnss3 libxcursor1 libasound2 libqt5gui5 libc++-dev libxcb-cursor0 htop tree tar gzip
```
```suggestion
RUN apt-get -o=Dpkg::Use-Pty=0 -q install -y ca-certificates curl git cpu-checker supervisor vim bash wget unzip xvfb x11vnc fluxbox xterm novnc net-tools htop libpulse-dev libnss3 libxcursor1 libasound2 libqt5gui5 libc++-dev libxcb-cursor0 htop tree tar gzip
```

- `apt-get` for non-interactive commands (`apt` is meant for humans, `apt-get` for scripts (better log output, less human-pretty formatting); also `apt` warns when used non-interactively).
- `-o=Dpkg::Use-Pty=0` so that interactive package configs can't stop and ask questions

jagerman commented on ci/rebuild-docker-images.py:
```diff
@@ -485,6 +488,98 @@ def playwright_build(distro, arch):
+WORKDIR /
+
+SHELL ["/bin/bash", "-c"]
+
```
It can be useful to throw this in here to avoid generating man pages during package installs/upgrades (which is kind of slow, and completely useless for a build container):
```
RUN /bin/bash -c 'echo "man-db man-db/auto-update boolean false" | debconf-set-selections'
```

jagerman commented on ci/rebuild-docker-images.py:
```diff
@@ -485,6 +488,98 @@ def playwright_build(distro, arch):
+        setuptools && \
+    pip3 install -U https://github.com/novnc/websockify/archive/refs/tags/v0.11.0.tar.gz
+
+RUN wget -O x11vnc.zip https://github.com/x11vnc/noVNC/archive/refs/heads/x11vnc.zip && \
```
Is there a specific reason for this version rather than https://github.com/novnc/noVNC?  The x11vnc version looks like someone's unmaintained personal fork.

jagerman replied:
Or even just `apt-get install novnc`

jagerman commented on ci/rebuild-docker-images.py:
```diff
@@ -485,6 +488,98 @@ def playwright_build(distro, arch):
+
+ENV HOME=/root \
+    LANG=en_US.UTF-8 \
+    DEBIAN_FRONTEN=noninteractive \
```
Missing a "D"

jagerman committed:
Various amendments

- Build on top of our debian-stable-base rather than the unmaintained
  `openjdk:...` docker images.

- Replace (unusable) docker ARGs with python variables

- Replace unzip with bsdtar piped extraction of command tools

- split into two: appium-34-base and appium-34-pixel6 (where either 34
  or pixel6 can easily be extended to build multiple devices as
  desired).

- use eatmydata for package install commands

jagerman committed:
Add --debug; use constant for apt-get invocation

--debug forces -j1, but prints output as it goes (rather than capturing
it and then showing a log if it fails).
